### PR TITLE
ne3np4 to namelist_defaults_ctsm.xml and Makefile for PTS mode and add ability for warm starts in PTS mode

### LINKED ISSUE #2768: Add ne3np4 to list of surface dataset resolutions
For support of SCAM we need to support ne3np4 in addition to ne3np4.pg3 as pointed out by @jtruesdal.

Definition of done:
- [x] Create the needed files from a branch tag of ctsm5.3.0 (do rest on b4b-dev)
- [x] Add build of ne3np4 for same configurations as ne3np4.pg3 (1850, 2000, Hist, SSP2-4.5) to mksurfdata_esmf Makefile
- [x] Add the files into namelist_defaults
- [x] Add a test for it to the build-namelist tester
- [x] Add a SCAM test using it to aux_clm (and ctsm_sci?)

### PR DESCRIPTION
**Pull request by slevis-lmwg**
### Description of changes
Same as title.

### Specific notes

Contributors other than yourself, if any:
@jtruesdal 

CTSM Issues Fixed (include github issue #):
Fixes #2768 

Are answers expected to change (and if so in what way)?
No.

Any User Interface Changes (namelist or namelist defaults changes)?
New fsurdat/landuse files added to the defaults.

Does this create a need to change or add documentation? Did you do so?
No.

Testing performed, if any:
On derecho
PASS Test the addition of the new fsurdat/landuse files to the Makefile with
```
make crop-global-SSP2-4.5-ne3
make crop-global-1850-low-res
make crop-global-present-low-res
```
PASS ./build-namelist_test.pl (before and after adding ne3np4 tests)
PASS python testing
PASS clm_pymods

### CODE REVIEW

**Comment by slevis-lmwg** [comment]
`./create_test SMS_D_Ld1_Mmpi-serial.ne3_ne3_mg37.I2000Clm50SpRs.derecho_intel.clm-ptsRLA`
fails with 
```
ERROR: surface dataset vs. land domain lon/lat mismatch error
   89.9999999999996       3.552713678800501E-014
```
which I think means that picked the wrong grid alias for ne3np4...

**Comment by slevis-lmwg** [comment]
- [ ] Rename files to the original names to reduce proliferation of files and to avoid anyone using the bad files.
- [ ] Copy to /inputdata and ./rimport
- [ ] Try the same tests

**Comment by slevis-lmwg** [comment]
The test still fails but with a new error. (Keep in mind that I named the new files the same as before.)

lnd.log stops here
```
Reading in urban input data from fsurdat file ...
(GETFIL): attempting to find local file 
surfdata_ne3np4_hist_2000_78pfts_c240925.nc
(GETFIL): using /glade/campaign/cesm/cesmdata/inputdata/lnd/clm2/surfdata_esmf/ctsm5.3.0/surfdata_ne3np4_hist_2000_78pfts_c240925.nc
 Opened existing file 
 /glade/campaign/cesm/cesmdata/inputdata/lnd/clm2/surfdata_esmf/ctsm5.3.0/surfda
 ta_ne3np4_hist_2000_78pfts_c240925.nc         489
 UrbanInput                      
 /glade/campaign/cesm/cesmdata/inputdata/lnd/clm2/surfdata_esmf/ctsm5.3.0/surfda
 ta_ne3np4_hist_2000_78pfts_c240925.nc
```
and cesm.log gives the error
```
Abort with message NetCDF: Start+count exceeds dimension bound in file /glade/derecho/scratch/csgteam/temp/spack/derecho/23.09/builds/spack-stage-parallelio-2.6.2-bpi7h2bnkshocep4hl3drfik4ebj44iu/spack-src/src/clib/pio_getput_int.c at line 770
...
cesm.exe           0000000003912EDB  urbanparamstype_m         572  UrbanParamsType.F90
cesm.exe           0000000000AA3953  clm_initializemod         235  clm_initializeMod.F90
```
@ekluzek I will follow up with John Truesdale in the issue. You thought maybe it was strange that the ne3np4 file had maxNodePElement = 5...

**Comment by slevis-lmwg** [comment]
@jtruesdal the test worked this time. Thank you for your help with this.

**Comment by slevis-lmwg** [comment]
From meeting with @ekluzek 
I should finalize testing and merge this PR.

**Comment by slevis-lmwg** [comment]
@jtruesdal 
I ran complete ctsm testing (details in my previous post, directly above) and one ptsRLA test failed on izumi-gnu, while the same test passed on derecho-intel. 

The FAIL on izumi: `/fs/cgd/data0/slevis/git_people/jtruesdal/ne3np4_to_defaults_and_makefile/tests_1108-124238iz/SMS_D_Ld1_Mmpi-serial.ne3_ne3_mg37.I2000Clm50SpRs.izumi_gnu.clm-ptsRLA.C.1108-124238iz_gnu`

The PASS on derecho:`/glade/derecho/scratch/slevis/tests_1108-123104de/SMS_D_Ld1_Mmpi-serial.ne3_ne3_mg37.I2000Clm50SpRs.derecho_intel.clm-ptsRLA.C.1108-123104de_int`

Izumi was upgraded recently. This has caused flakiness in our Mmpi-serial tests. Still, this particular error looks different, so I wonder whether it's real, despite not failing on derecho.

If you feel we can disregard this failure on izumi, then I would ask you to review/approve my PR, so that we may merge it. Again, thank you for your help, John.

**Comment by jtruesdal** [comment]
If you've run it a few times and it fails in the same way I think I'll have to look into more.  It appears to be dying in the land and could be related to my mods.

**Comment by slevis-lmwg** [comment]
@jtruesdal the test's /run directory shows identical failure results from three attempts. (Clm's izumi test-suite runs multiple attempts automatically to get past machine flakiness.)

**Comment by jtruesdal** [comment]
@ekluzek I'm looking at the test in question but am unsure how the land pts mode test is supposed to work? I always thought the test was setup with boundary data containing just a point but this isn't the case as its using full resolution fsurfdat etc.  Did this test run with previous versions of the SE grid? If you can point me to a working test I can see what is going wrong with this new SE grid.

**Comment by slevis-lmwg** [comment]
I take it back, I submitted the wrong test. Let me try again.

**Comment by slevis-lmwg** [comment]
The test still fails. Latest one is here on izumi:
`/scratch/cluster/slevis/SMS_D_Ld1_Mmpi-serial.ne3_ne3_mg37.I2000Clm50SpRs.izumi_gnu.clm-ptsRLA.20241113_163638_5fyem2`

For a test that works, how about the derecho equivalent of this one:
`/glade/derecho/scratch/slevis/tests_1108-123104de/SMS_D_Ld1_Mmpi-serial.ne3_ne3_mg37.I2000Clm50SpRs.derecho_intel.clm-ptsRLA.C.1108-123104de_int`

I don't have other such tests using a SE grid. Prior to using the ne3 grid, we ran these tests with f45. For example:
`/glade/derecho/scratch/slevis/tests_1113-122038de/SMS_D_Ld1_Mmpi-serial.f45_f45_mg37.I2000Clm50SpRs.derecho_intel.clm-ptsRLA.GC.1113-122038de_int`

**Comment by ekluzek** [comment]
Actually, @jtruesdal and @slevis-lmwg I think this is likely something we may need to talk about in a meeting. PTS_MODE for CTSM in the past was to run CLM how it would run within SCAM -- but in an I case. It's primary purpose being to make sure we don't screw up CLM with our changes, but it's also something that *could* be useful for CLM I cases on their own.

So it was setup to read in full 2D grids and pick a point out of that 2D grid. But, I think it might have assumed that the 2D grid is a regular grid and not in vector format (like the SE grids). If so I'm thinking that PTS_MODE might not work for SE grids. But, is SCAM updated on the CAM side to use unstructured grids like for the SE dycore? If so we might need to do the same thing in CTSM.

In any case we might need to talk more about this to make sure we know what we need to support SCAM on the CLM side of things. So we'll need to know how SCAM currently works within CAM and how you need it to work.

**Comment by jtruesdal** [comment]
@slevis-lmwg @ekluzek I was thinking along the lines of Erik as the code I looked at doesn't appear to have the logic to handle an unstructured grid in PTS mode.  The derecho test has warning messages in the log about not being able to find a proper lat/lon column.  I think it is running but giving wrong results.  This I test also looks like it is trying to interpolate an f09 initial condition for a single column which would require my warm start mods PR.  I can retest this case after a test merge of those mods to see if they fix the issue.  My guess is that there might be a few more minor mods to get it running but shouldn't be hard.  I can do that work but it will take a few days for me to get to it.

**Comment by slevis-lmwg** [comment]
@jtruesdal 
Heads up that @ekluzek suggested the three of us meet some time next week to discuss this. I will find a time that works for us.

**Comment by slevis-lmwg** [comment]
@jtruesdal I confirmed that the last commit in this PR
https://github.com/ESCOMP/CTSM/pull/2835/commits/a389e6dac1069807540f868d79bdcad36fdb0787
is the same as what "git log" shows in my local branch here:
`/glade/work/slevis/git_people/jtruesdal/ne3np4_to_defaults_and_makefile`

I will wait until after you finish your part of this to update to the latest `cesm3_0_beta04_changes` branch.

**Comment by jtruesdal** [comment]
Thanks @slevis-lmwg I created a PR against your branch

**Comment by slevis-lmwg** [comment]
Sounds good @jtruesdal 
When it's ready for me to merge back to this PR, pls let me know the link to your PR. 
You should also have appropriate permissions to merge it back here yourself, if you prefer to do that.

**Comment by ekluzek** [comment]
@slevis-lmwg and @jtruesdal bringing this up again, since we are all back working after the holidays. @slevis-lmwg has another task that proceeds this in priority. But, this looks like something simple enough to come in.

@jtruesdal from above it looks like you still need to merge your PR into this one? Is that right? And if so could you do that? Thanks.

**Comment by jtruesdal** [comment]
Hi Erik:

Thanks for circling back. I will have time to test and merge my PR
tomorrow.  I'll ping you two when it's done.

jt

**Comment by jtruesdal** [comment]
@ekluzek I've completed testing of the warm start scam CLM mods.  After some debugging and making a bug fix, I've verified that the SCAM column resultes are bit for bit identical with a column run in the global model.  I've asked @slevis-lmwg to check the standard CLM regression tests to make sure I didn't break anything before merging my PR into his ne3np4_to_defaults_and_makefile branch.  I will make a parallel commit to CAM which has the CAM warm start regression test for SCAM which will be a required part of our testing and checks bit for bit results of a CAM/CLM SCAM warm start run.  That parallel CAM commit needs to go in after this PR is merged.

**Comment by slevis-lmwg** [comment]
@jtruesdal I'm happy to start the aux_clm test-suite and/or any other testing that @ekluzek may recommend. Please let me know where I can find your branch so I can do a git checkout.

**Comment by slevis-lmwg** [comment]
@jtruesdal sorry, I just saw the separate notification from https://github.com/slevis-lmwg/ctsm/pull/17

File: src/init_interp/initInterp.F90 [hunk 1]
```diff
@@ -942,35 +943,55 @@ subroutine set_subgrid_info(beg, end, dimname, use_glob, ncid, active, subgrid)
...
       ! form of ncd_io
       type(file_desc_t)  , intent(inout) :: ncid
       character(len=*)   , intent(in)    :: varname
       real(r8), pointer  , intent(inout) :: data(:)
       character(len=*)   , intent(in)    :: dim1name
       logical            , intent(in)    :: use_glob  ! if .true., use the 'glob' form of ncd_io
+      logical            , intent(in)    :: allow_scm ! if .true., use the 'glob' form of ncd_io
+
+      ! local
+      character(16)                      :: readflag
```
Comment (ekluzek) [inline src/init_interp/initInterp.F90:957]: We should add some comments here to explain what's going on with this scm/noscm stuff.

**Review by ekluzek — APPROVED** [review]
@slevis-lmwg and I looked at this some. This has the changes from #2333 so that PR will likely become smaller when this comes in. There still might be something there, and I added a couple things to do there also.

I wasn't sure about all of the allow_scm logic with read and read_noscm and the like. But, I've come around to it. I think all of this is required. So the only thing might be changing the names. But, I think it's clear enough that SCM means Single Column Model, although I do think there might be places where some comments around this might be helpful.

I'm adding approve even with the request for a few comments, just so you don't have to have me reapprove. Anything needed to be addressed could go into #2333.

**Comment by slevis-lmwg** [comment]
derecho
OK make black and lint (the latter gives two warnings)
PASS python
PASS ./build-namelist_test.pl
IN PROG `./run_sys_tests -s aux_clm -c ctsm5.3.020 --skip-generate`

izumi
waiting for machine to return to svc

File: src/init_interp/initInterp.F90 (hunk 1 above)
Comment (slevis-lmwg) [reply src/init_interp/initInterp.F90:957]: Looking through this a bit, the comment right after `allow_scm` is an accidental copy from line 953 (directly above it). As such, I will replace the accidental comment with the one that I see elsewhere:
`      logical            , intent(in)    :: allow_scm ! if .true., allow scm data slice`

Comment (slevis-lmwg) [reply src/init_interp/initInterp.F90:957]: I found an even better instance of the comment explaining `allow_scm`. Now using that one.

Comment (jtruesdal) [reply src/init_interp/initInterp.F90:957]: Thanks Sam.  The comment should be changed as you recommend.  As for the read_noscm flag this is to distinguish those times during IC interpolation when you are running in single column mode but need to read the full data grid.  Normally single_column means read the data grid and extract the closest column but during IC interpolation you need to read in the full grid to be interpolated regardless of the single_column flag.

**Comment by slevis-lmwg** [comment]
I found I had to merge the latest https://github.com/ESCOMP/CTSM/tree/cesm3_0_beta04_changes to this PR's branch. The diffs pertain to CONUS, POLARCAP, and ARCTIC* ctsm_sci tests. I see these as orthogonal to this PR's work, plus ctsm_sci was not part of this PR's testing to begin with. So I will proceed with the merge.

**Comment by slevis-lmwg** [comment]
@ekluzek I wrapped this up just now. I tried `git tag | grep cesm` to see if I should make a tag of it, but it doesn't look as though we made any tags in the cesm3_0_beta04_changes branch previously.
